File: web/src/views/private/Cartilla/validations/cartilla.ts
import * as Yup from "yup";
// Esquema de validación con Yup
export const cartillaSchema = Yup.object({
  // Datos de la Madre
  nombreMadre: Yup.string().required("Nombre de la madre es requerido"),
  ciMadre: Yup.string().required("CI de la madre es requerido"),
  direccionMadre: Yup.string().required("Dirección de la madre es requerida"),
  detalleDireccionMadre: Yup.string(),
  telefonoMadre: Yup.string().required("Teléfono de la madre es requerido"),
  telefonoEmergenciaMadre: Yup.string().required(
    "Teléfono de emergencia es requerido"
  ),
  tratamientoHiportiroidismo: Yup.boolean().required(
    "El tratamiento de hipotiroidismo es requerido"
  ),
  tratamientoHipertiroidismo: Yup.boolean().required(
    "El tratamiento de hipertiroidismo es requerido"
  ),
  tratamientoMadre: Yup.string().required(
    "Tratamiento de la madre es requerido"
  ),
  enfermedadMadre: Yup.string().required("Enfermedad de la madre es requerida"),
  idProvincia: Yup.number().nullable().required("Provincia es requerida"),

  // Datos del Paciente
  nombrePaciente: Yup.string().required("Nombre del paciente es requerido"),
  sexoPaciente: Yup.string().required("Sexo del paciente es requerido"),
  edadGestacionalSemanaPaciente: Yup.number().required(
    "Edad gestacional (semanas) es requerida"
  ),
  edadGestacionalDiaPaciente: Yup.number().required(
    "Edad gestacional (días) es requerida"
  ),
  fechaNacimientoPaciente: Yup.date().required(
    "Fecha de nacimiento es requerida"
  ),
  pesoNacimientoPaciente: Yup.number().required(
    "Peso de nacimiento es requerido"
  ),
  nacimientoTerminoPaciente: Yup.boolean().required(
    "Indicación de nacimiento a término es requerida"
  ),

  // Datos de la Cartilla
  codigoBarras: Yup.string().required("Código de barras es requerido"),
  fechaTomaMuestras: Yup.date().required(
    "Fecha de toma de muestras es requerida"
  ),
  numeroMuestra: Yup.number().required("Número de muestra es requerido"),
  transfucion: Yup.boolean().required("Indicación de transfusión es requerida"),
  antibioticos: Yup.string().nullable(),
  notas: Yup.string().nullable(),
});

// Definición de la interfaz para el formulario usando Yup.InferType
export interface CartillaForm extends Yup.InferType<typeof cartillaSchema> {}

File: server/Dtos/Cartilla.cs
namespace server.Dtos
{
    public class CartillaDTO
    {
        // Datos de la Madre
        public required string NombreMadre { get; set; }
        public required string CiMadre { get; set; }
        public required string DireccionMadre { get; set; }
        public required string? DetalleDireccionMadre { get; set; }
        public required string TelefonoMadre { get; set; }
        public required string TelefonoEmergenciaMadre { get; set; }
        public required bool TratamientoHiportiroidismo { get; set; }
        public required bool TratamientoHipertiroidismo { get; set; }
        public required string TratamientoMadre { get; set; }
        public required string EnfermedadMadre { get; set; }
        public required int? IdProvincia { get; set; }

        // Datos del Paciente
        public required string NombrePaciente { get; set; }
        public required string SexoPaciente { get; set; }
        public required int EdadGestacionalSemanaPaciente { get; set; }
        public required int EdadGestacionalDiaPaciente { get; set; }
        public required DateOnly FechaNacimientoPaciente { get; set; }
        public required double PesoNacimientoPaciente { get; set; }
        public required bool NacimientoTerminoPaciente { get; set; }

        // Datos de la Cartilla
        public required string CodigoBarras { get; set; }
        public required DateOnly FechaTomaMuestras { get; set; }
        public required int NumeroMuestra { get; set; }
        public required bool? Transfucion { get; set; }
        public required string? Antibioticos { get; set; }
        public required string? Notas { get; set; }
    }
}